src/Analysis/AtlasDataDictionary/AtlasDataDictionaryContainer.tsx

```diff
@@ -1,73 +1,123 @@
-    sortKey: null,
-    direction: 'off',
-  });
 
-  /* Pagination */
-  const [entriesShown, setEntriesShown] = useState(10);
-  const [activePage, setActivePage] = useState(1);
-  const paginatedData = data.slice(
-    entriesShown * activePage - entriesShown,
-    entriesShown * activePage,
+  const [dataDictionaryTableState, setDataDictionaryTableState] = useState(
+    InitialDataDictionaryTableState,
   );
+  const {
+    openDropdowns,
+    searchTerm,
+    sortConfig,
+    currentPage,
+    entriesShown,
+    columnsShown,
+  } = dataDictionaryTableState;
 
-  const rows = paginatedData.map((rowObject, i) => (
-    <TableRow
-      key={i}
-      rowObject={rowObject}
-      columnsShown={columnsShown}
-      searchInputValue={searchInputValue}
-    />
-  ));
+  const entriesHeaderStart = dataDictionaryTableState.entriesShown
+      * dataDictionaryTableState.currentPage
+    - dataDictionaryTableState.entriesShown
+    + 1;
+
+  const entriesHeaderStop = dataDictionaryTableState.entriesShown
+    * dataDictionaryTableState.currentPage;
+
+  const paginatedData = data.slice(
+    entriesShown * currentPage - entriesShown,
+    entriesShown * currentPage,
+  );
 
-  const handleSort = (sortKey) => {
-    let direction: ISortConfig['direction'] = 'ascending';
-    if (sortConfig.sortKey === sortKey) {
-      if (sortConfig.direction === 'ascending') {
-        direction = 'descending';
-      } else if (sortConfig.direction === 'descending') {
-        direction = 'off';
-      } else if (sortConfig.direction === 'off') {
-        direction = 'ascending';
-      }
+  const handleTableChange = (
+    event:
+      | 'openDropdown'
+      | 'closeDropdown'
+      | 'currentPage'
+      | 'entriesShown'
+      | 'searchTerm'
+      | 'sortConfig'
+      | 'columnManagement',
+    eventData: any,
+  ) => {
+    if (event === 'openDropdown') {
+      setDataDictionaryTableState({
+        ...dataDictionaryTableState,
+        openDropdowns: [...openDropdowns, eventData],
+      });
+    } else if (event === 'closeDropdown') {
+      setDataDictionaryTableState({
+        ...dataDictionaryTableState,
+        openDropdowns: openDropdowns.filter(
+          (dropdownNumber: number) => dropdownNumber !== eventData,
+        ),
+      });
+    } else if (event === 'currentPage') {
+      setDataDictionaryTableState({
+        ...dataDictionaryTableState,
+        currentPage: eventData,
+      });
+    } else if (event === 'entriesShown') {
+      setDataDictionaryTableState({
+        ...dataDictionaryTableState,
+        entriesShown: eventData,
+        currentPage: 1,
+      });
+    } else if (event === 'searchTerm') {
+      setDataDictionaryTableState({
+        ...dataDictionaryTableState,
+        searchTerm: eventData,
+        currentPage: 1,
+      });
+    } else if (event === 'sortConfig') {
+      setDataDictionaryTableState({
+        ...dataDictionaryTableState,
+        sortConfig: eventData,
+        currentPage: 1,
+      });
```

Comment:
can we perhaps combine these? They are doing the same thing here.

Reply:
Great idea! Updated to use square brackets so the event variable is evaluated as the key of the key value pair assignment: 
```
} else if (event === 'entriesShown' || event === 'searchTerm' || event === 'sortConfig') {
      setDataDictionaryTableState({
        ...dataDictionaryTableState,
        [event]: eventData,
        currentPage: 1,
      });
```